src/main/groovy/life/qbic/datamodel/datasets/datastructure/folders/nanopore/UnclassifiedFastQFolder.groovy

```diff
@@ -0,0 +1,58 @@
+package life.qbic.datamodel.datasets.datastructure.folders.nanopore
+
+import life.qbic.datamodel.datasets.datastructure.files.nanopore.FastQFile
+import life.qbic.datamodel.datasets.datastructure.files.nanopore.FastQZippedFile
+import life.qbic.datamodel.datasets.datastructure.folders.DataFolder
+/**
+ * A data folder, which name contains `unclassified` read information and contains only fastq files.
+ *
+ * This class should be used, when there is unclassified read information available in
+ * an Oxford Nanopore sequencing measurement. Unclassified means, that the read barcode used to
+ * determine the sample origin was unknown or missing.
+ *
+ * The children of this folder are always of type FastQFile.
+ *
+ * The name of an unclassified folder type must contain the substring `unclassified` in order
+ * to get determined as such.
+ *
+ * @author Sven Fillinger
+ * @since 1.9.0
+ */
+class UnclassifiedFastQFolder extends DataFolder {
+
+    final private static String NAME_SCHEMA = /unclassified/
+
+    protected UnclassifiedFastQFolder() {}
+
+    protected UnclassifiedFastQFolder(String name, String relativePath, List children) {
+        super(name, relativePath, children)
+        validateName()
+    }
+    /**
+     * Creates a new instance of a UnclassifiedFastQFolder object
+     * @param relativePath The relative path of the folder
+     * @param children A list with FastQFile or FastQZipped elements of the folder
+     * @return A new instance of a UnclassifiedFastQFolder object
+     */
+    static UnclassifiedFastQFolder create(String name, String relativePath, List<?> children) {
+        if (children) {
+            validateChildren(children)
+        }
+        new UnclassifiedFastQFolder(name, relativePath, children)
+    }
+
+    private static void validateChildren(List children) {
+        if (!((children.get(0) instanceof FastQFile) || (children.get(0) instanceof FastQZippedFile))) {
```

Comment:
Is it enough to only test the first element of the children list?

Reply:
True, it is better to iterate. I will fix this.

Reply:
https://github.com/qbicsoftware/data-model-lib/pull/26/commits/1929e73d0d8693b54890784834966e5deba3571f